### ISSUE TITLE
When we use test_slide to test, but can't get the result like  glasses in Fig. 7.

### ISSUE THREAD

**Issue by c1a1o1** [elvisyjlin/AttGAN-PyTorch#14]
When we tested with test_slide, we couldn't get the glasses results shown in Figure 7 in your paper.

Can you give some advice?
![182639](https://user-images.githubusercontent.com/7991905/56109752-81d43280-5f84-11e9-9a71-5aa35be74787.jpg)
![182653](https://user-images.githubusercontent.com/7991905/56109761-8993d700-5f84-11e9-88af-7ffd2aca1234.jpg)

It can be seen that it is impossible to generate dark glasses

**Comment by elvisyjlin** [comment]
First, I'm not the author of the paper. I'm doing research in face editing so I implemented AttGAN in PyTorch.

To be honest, the model does not perform well on all images. As you can see in my custom results, some faces look not good.

The glasses are not quite visible in your images. My suggestion is that you can try with a larger test value rather than just 1. For example, test with `--test_int_max 1.5`, you will probably get better glasses. On the other hand, you mentioned that you'd like to generate sunglasses. The pretrained model and the implemented code does not support it because the labels in CelebA are like `with glasses / without glasses`. This code generates a face with desired attributes, but we are not able to constraint the style of each attribute. If you want to control the style, you have to implement the __Mutual Information Maximization__ as the paper describes. I have no plan to do it so far.

**Comment by c1a1o1** [comment]
Thank you very much!  Maybe you can look at this work for face editing!
https://github.com/quolc/neural-collage

**Comment by elvisyjlin** [comment]
This work is pretty cool! I'll read it. Thank you for the suggestion!

**Comment by c1a1o1** [comment]
I hope you can use pytorch to reproduce it, thank you.
